eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ServiceInterfaceCheck.java

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * The @ServiceInterface class should have the following rules:
+ *   1) The annotation property 'name' should be non-empty
+ *   2) The length of value of property 'name' should be less than 10 characters and without space
+ */
+public class ServiceInterfaceCheck extends AbstractCheck {
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.INTERFACE_DEF
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.INTERFACE_DEF:
+                checkServiceInterface(token);
+                break;
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     *  The @ServiceInterface class should have the following rules:
+     *   1) The annotation property 'name' should be non-empty
+     *   2) The length of value of property 'name' should be less than 10 characters and without space
+     *
+     * @param interfaceDefToken INTERFACE_DEF AST node
+     */
+    private void checkServiceInterface(DetailAST interfaceDefToken) {
+        DetailAST serviceInterfaceAnnotationNode = null;
+        String nameValue = null;
+
+        DetailAST modifiersToken = interfaceDefToken.findFirstToken(TokenTypes.MODIFIERS);
+        // Find the @ServiceInterface and the property 'name' and the corresponding value
+        for (DetailAST ast = modifiersToken.getFirstChild(); ast != null; ast = ast.getNextSibling()) {
```

Comment:
Consider extracting some parts of this for loop into smaller methods for better readability.